### DISCUSSION TITLE
RTPS Relay Tool

### DISCUSSION THREAD

**Discussion by amielc1** [OpenDDS/OpenDDS#4668] [Q&A] [answered]
Hi, 
I need to use the Rtps Realy tool for Discovery via Unicast. 
When I tried to compile OpenDDS with opendds_sucurety - to get RtpsTool Libs - I got a lot of errors.
1. Can I get an instruction for compiling Rtps Tool? 
2. Can I get a  compiled RtpsTool?  

Thank a lot

**Comment by jrw972** [comment]
A Docker image is [available](https://github.com/OpenDDS/OpenDDS/pkgs/container/opendds).

The [Dockerfile](https://github.com/OpenDDS/OpenDDS/blob/master/Dockerfile) shows how to build with security.
Many of the [GitHub Actions jobs](.github/workflows/build_and_test.yml) also show how to build with security.

**Reply by amielc1** [reply, marked as answer]
thank you, @jrw972 .
I need to run Rtps relay outside from docker. 
locally on my computer. do you have any advice?

**Reply by jrw972** [reply]
If you build OpenDDS with security, the RtpsRelay will be built.

See https://opendds.readthedocs.io/en/latest-release/devguide/building/index.html for instruction on building OpenDDS in general and with security.

You can use the Dockerfile and GitHub Actions workflows which show one way of installing the necessary dependencies and building with ``--security``.
